### ISSUE TITLE
Analyzer should stop depending on deprecated code from package:package_config

### ISSUE THREAD

**Issue by lrhn** [dart-lang/sdk#44116]
The analyzer depends on `package:package_config/packages_file.dart`, which is deprecated.
Example from: analyzer/lib/src/context/packages.dart:
```dart
// ignore: deprecated_member_use
import 'package:package_config/packages_file.dart' as dot_packages;
```
Ignoring deprecation does not stop it being deprecated (or going away). The deprecated code will be removed in the 2.0.0 release of `package_config`.

The analyzer should either migrate to using `package_config/package_config.dart`, if possible, or copy the code they use into the analyzer before it's removed.

**Comment by lrhn** [comment]
(Now blocking https://github.com/dart-lang/package_config/pull/93).

**Comment by natebosch** [comment]
Looks like this is blocked by https://github.com/dart-lang/package_config/issues/95

**Comment by scheglov** [comment]
Sam has work in progress https://dart-review.googlesource.com/c/sdk/+/171880

**Comment by lrhn** [comment]
That change is not a fix. I'll revert it.

**Comment by srawlins** [comment]
That sounds good. I think that's a new issue. Analyzer no longer depends on deprecated code from package:package_config.

**Comment by devoncarew** [comment]
Dropping to a P2; I don't believe this is blocking a null safe release of package_config anymore. Please let us know if you think this is not the case.

**Comment by jakemac53** [comment]
Removing a private import to a dependency should be high priority as well, its more dangerous and a worse situation to be in than depending on a deprecated api.

**Comment by jakemac53** [comment]
@lrhn was the null safe version of this package also planning on removing support for `.packages` since it would be a breaking release?

**Comment by lrhn** [comment]
Sadly not, that seems to be too soon since we haven't actually stopped depending on `.packages`  everywhere yet.
(I'd love to remove that support ASAP, but the first step would be for `pub` to stop generating the file, and that hasn't happened yet).

**Comment by parlough** [comment]
Fixed by https://github.com/dart-lang/sdk/commit/f9d644355718e2840beac65f2779414e690f2d42#diff-e598e571cedc0765cae1c9a03fe410583daadd6d50c0583e3cdfd98f35429a3cL8
